src/OrchardCore.Modules/OrchardCore.Facebook/Endpoints/Api/InitSdkEndpoint.cs

```diff
@@ -0,0 +1,56 @@
+using System.Globalization;
+using Microsoft.AspNetCore.Routing;
+using Microsoft.AspNetCore.Authorization;
+using Microsoft.AspNetCore.Builder;
+using Microsoft.AspNetCore.Http;
+using OrchardCore.ContentManagement;
+using OrchardCore.Settings;
+using OrchardCore.Facebook.Settings;
+
+namespace OrchardCore.Facebook;
+internal static class InitSdkEndpoint
+{
+    public static IEndpointRouteBuilder AddInitSdkEndpoint(this IEndpointRouteBuilder builder)
+    {
+        builder.MapGet("orchardcore/facebook/sdk/fbsdk.js", HandleAsync)
+            .AllowAnonymous()
+            .DisableAntiforgery();
+
+        return builder;
+    }
+
+    [Authorize(AuthenticationSchemes = "Api")]
+    private static async Task<IResult> HandleAsync(
+        ContentItem model,
+        ISiteService siteService, 
+        HttpContext httpContext,
+        bool draft = false)
+    {
+        ArgumentNullException.ThrowIfNull(httpContext);
+
+        var script = await getInitSdkScript(siteService);
+
+        if (script == null)
+        {
+            return Results.NotFound(); 
+        }
+
+        return Results.Ok(script); 
```

Comment:
Could this return a utf8 encoded `byte[]` ?

Reply:
If `Results` is not powerful enough , maybe use the response writer directly (PipeWriter?, TextWriter?).